src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs

```diff
@@ -547,48 +450,22 @@ public bool IsSubstringAt(int index, Utf8String s)
             return Substring(index, s.Length).Equals(s);
         }
 
-        public void CopyTo(ByteSpan buffer)
+        public void CopyTo(Span<byte> buffer)
         {
-            if(buffer.Length < Length)
+            if (buffer.Length < Length)
             {
                 throw new ArgumentException("buffer");
             }
 
-            if(_bytes == null)
-            {
-                _buffer.TryCopyTo(buffer);
-            }
-            else
+            if (!_buffer.TryCopyTo(buffer))
             {
-                unsafe {
-                    fixed(byte* pBytes = _bytes) {
-                        buffer.TrySet(pBytes, _bytes.Length);
-                    }
-                }
+                throw new Exception("Internal error: range check already done, no errors expected here");
```

Comment:
True but in this case if this exception is thrown that means that the library is broken so handling it is futile. I'd like to have error handling review separately - this one is solely for conversion between bytes, I don't want to mix refactoring in it